.github/workflows/backwards_compatibility_marqo_orchestrator.yml

```diff
@@ -0,0 +1,176 @@
+    needs: check-if-image-exists
+    if: needs.check-if-image-exists.outputs.image_exists == 'false'
+    uses: ./.github/workflows/build_push_img.yml
+    secrets: inherit
+    with:
+      marqo_ref: "${{ github.sha }}"
+      push_to: "ECR"
+      image_repo: "marqo-compatibility-tests"
+      image_tag: "${{ github.sha }}"
+
+
+  orchestrate:
+    # Job to orchestrate backwards compatibility test execution. Majorly responsible for determining to_version and for generating the list of for_version to test against.
+    name: Orchestrate backwards compatibility test execution
+    runs-on: ubuntu-latest
+    needs: build-and-push-image
+    outputs:
+      list: ${{ steps.generate-versions.outputs.list }}
+      to_version: ${{ steps.get-to-version.outputs.to_version }}
+      image_digest: ${{ steps.image-digest.outputs.digest }}
+    environment: marqo-test-suite
+    steps:
+      # Step to check out the Marqo repository
+      - name: Checkout marqo repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      # Step to set up Python 3.9
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3.9'
+          cache: "pip"
+
+      # Step to install the semver package
+      - name: Install semver
+        run: |
+          pip install semver
+
+      # Step to determine the target version
+      - name: Determine to_version
+        id: get-to-version
+        run: |
+          if [ "${{ github.event_name }}" == "workflow_dispatch" ] && [ -n "${{ github.event.inputs.to_version }}" ]; then
+            VERSION="${{ github.event.inputs.to_version }}"
+          else
+            VERSION=$(python tests/backwards_compatibility_tests/scripts/determine_to_version.py ${{ github.sha }})
+          fi
+          echo "::set-output name=to_version::$VERSION"
+
+      # Step to generate the list of versions to test
+      - name: Generate version list #this code block just generates the from_version list and stores it in a versions variable as a list
+        id: generate-versions
+        run: |
+          # Run the Python script and capture its output
+          VERSION_LIST=$(python tests/backwards_compatibility_tests/scripts/generate_versions.py ${{ steps.get-to-version.outputs.to_version }} ${{ env.MAX_VERSIONS_TO_TEST }})
+          echo "::set-output name=list::$VERSION_LIST" 
+
+      # Step to display the versions to test
+      - name: display versions
+        run: |
+          echo "Versions to test: ${{ steps.generate-versions.outputs.list }} against to_version: ${{ steps.get-to-version.outputs.to_version }}"
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id: ${{ secrets.ECR_PUSHER_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ECR_PUSHER_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Retrieve image digest
```

Comment:
See comment above on where this digest must come from

Reply:
This now accepts a digest - I have completely removed the whole logic to retrieve digest using tags

1. https://github.com/marqo-ai/marqo/commit/740097f76bc5fd67568d409c1a680d9ebcb77333#diff-f4c34005d6dcd764124973ffb9f157dd84675dec760dc24c981a4d02fec8a9daL134-L162
2. https://github.com/marqo-ai/marqo/commit/740097f76bc5fd67568d409c1a680d9ebcb77333#diff-a60798b1e43d7cc0bb54656f01ca73fc84579b6e501acc45d79cbfec3819cb1dR25-R28